### ISSUE TITLE
VoiceOver is not announcing the index of options inside a mat autocomplete. 

### ISSUE THREAD

**Issue by bq007** [angular/material#11950]
## Bug, enhancement request, or proposal

**Demo URL (required)***:  
Angular Material demos for mat autocomplete. 
https://material.angular.io/components/autocomplete/examples

#### Detailed Reproduction Steps
1. Enable VoiceOver screen reader. 
2. Open the angular demo for mat-autocomplete in angular material in chrome. https://material.angular.io/components/autocomplete/examples
3. Pick one of the examples and trigger the autocomplete option drop-list.
4. Use the keyboard to move between different options and listen to screen reader. 

### Explain the expected behavior
The VoiceOver screen reader should announce the index of each option. 

### Explain the current behavior
The VoiceOver screen reader only announces the option label + "clickable" without index. 

### Discuss the use-case or motivation for changing the existing behavior
This feature to impacts user accessibility experience. 

### List the affected versions of AngularJS, Material, OS, and browsers

- AngularJS: 10.0.0
- AngularJS Material: 10.0.0
- OS: macOS Catalina Version 10.15.3
- Browsers: Chrome Version 83.0.4103.116 (Official Build) (64-bit)
 
### Add anything else we should know
Adding roles described in 
https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.0pattern/combobox-autocomplete-both.html does not fix the issue.

**Comment by Splaktar** [comment]
Please submit Angular Material and CDK questions [here](https://groups.google.com/forum/#!forum/angular-material2) and issues [here](https://github.com/angular/components/issues). This repo is for AngularJS Material.

I have transferred this issue to the correct repository for you.
